### ISSUE TITLE
TypeError: Cannot assign to read only property 'area' of object '#<Object>'

### ISSUE THREAD

**Issue by vaynevayne** [mantinedev/mantine#4500]
### What package has an issue

@mantine/core

### Describe the bug

<img width="1112" alt="image" src="https://github.com/mantinedev/mantine/assets/124666577/0f8e7f5a-a023-461d-954d-0544952235d6">

Can you see the bug?

maybe js disables modifying const objects
```jsx
const cloned = klona(values); //  Can we solve this problem by replacing it with let?
 cloned[splittedPath[0]] = value;
```

### What version of @mantine/hooks page do you have in package.json?

latest

### Do you know how to fix the issue

None

### Are you willing to participate in fixing this issue and create a pull request with the fix

None

**Comment by rtivital** [comment]
Please send a sandbox that reproduces the issue, template –https://codesandbox.io/s/mantine-template-9ze89?file=/src/App.tsx:0-6

**Comment by ChiFangChen** [comment]
do you use something like redux to deal with your initial value?
maybe you can try to destructure the object by this:
```
    const form = useForm({
        initialValues: { ...valueFromReduxOrImmutableValue },
    });
```

**Comment by vaynevayne** [comment]
no, i don't use any lib, just do this
```jsx
 const form = useForm({
        initialValues: { ...props.form },
    });
```

**Comment by rtivital** [comment]
This issue is closed for one of these reasons:

- It was fixed in 7.0
- It is no longer reproducible in 7.0
- It is no longer applicable in 7.0
- It does not have a reproduction

If you think that this issue was closed by mistake, and it is still an issue in version 7.0, please reopen it.
